pilot/pkg/networking/core/v1alpha3/listenermatch.go

```diff
@@ -0,0 +1,86 @@
+package v1alpha3
+
+import (
+	xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+
+	networking "istio.io/api/networking/v1alpha3"
+)
+
+func findMatches(listenerMatch *networking.EnvoyFilter_ListenerMatch, builder *ListenerBuilder) []*xdsapi.Listener {
+	var matches []*xdsapi.Listener
+	listeners := builder.getListeners()
+
+LISTENER_MATCH:
+	for _, listener := range listeners {
+		if listenerMatch.GetName() != "" && listenerMatch.GetName() != listener.GetName() {
+			continue
```

Comment:
we should treat name match as terminal. Once name matches, then nothing else matters as all listeners have unique names.